### PR TITLE
Add initial project files

### PR DESCRIPTION
**Pull request by TheMeinerLP**
These files were extracted from the main repository for the purpose of separate updating and maintaining.

### CODE REVIEW

File: sign-core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>SignSystem</artifactId>
+        <groupId>eu.cloudnetservice</groupId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+
```
Comment (GiantTreeLP) [inline sign-core/pom.xml:10]: you are missing a groupId here

File: sign-core/src/main/java/eu/cloudnetservice/sign/core/models/Position.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package eu.cloudnetservice.sign.core.models;
+
+import java.util.Objects;
+
+public final class Position {
+
+	private final double x,y,z;
+	private final String world,group;
```
Comment (GiantTreeLP) [inline sign-core/src/main/java/eu/cloudnetservice/sign/core/models/Position.java]: please don't use these constructs, declare only one field per line.

File: sign-core/src/main/java/eu/cloudnetservice/sign/core/models/Position.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		return z;
+	}
+
+	public String getWorld() {
+		return world;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
```
Comment (GiantTreeLP) [inline sign-core/src/main/java/eu/cloudnetservice/sign/core/models/Position.java]: please use the IDE generator for equals and hashcode

File: sign-core/src/main/java/eu/cloudnetservice/sign/core/models/SignLayout.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package eu.cloudnetservice.sign.core.models;
+
+public final class SignLayout {
+
+	/**
+	 * blockIds are not supported in all versions, use {@link SignLayout#blockName} instead
+	 */
+	@Deprecated
+	int blockId;
```
Comment (GiantTreeLP) [inline sign-core/src/main/java/eu/cloudnetservice/sign/core/models/SignLayout.java]: why is this not private?

File: sign-core/src/main/java/eu/cloudnetservice/sign/core/packets/PacketInSignSelector.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.stream.Collectors;
+
+public final class PacketInSignSelector extends PacketInHandler {
+	@Override
+	public void handleInput(Document data, PacketSender packetSender) {
+
+		Map<UUID, Sign> signMap = data.getObject("signs", TypeToken.getParameterized(Map.class, UUID.class, Sign.class).getType());
+		SignLayoutConfig signLayoutConfig = data.getObject("signLayoutConfig", TypeToken.get(SignLayoutConfig.class).getType());
+
+		Map<UUID, Sign> values = signMap.entrySet().stream().filter(uuidSignEntry -> uuidSignEntry.getValue().getPosition().getGroup().equals(CloudAPI.getInstance().getGroup())).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
Comment (GiantTreeLP) [inline sign-core/src/main/java/eu/cloudnetservice/sign/core/packets/PacketInSignSelector.java]: please use linebreaks here

File: sign-core/src/main/java/eu/cloudnetservice/sign/core/packets/PacketInSignSelector.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		if (SignManager.getInstance() != null) {
```
Comment (GiantTreeLP) [inline sign-core/src/main/java/eu/cloudnetservice/sign/core/packets/PacketInSignSelector.java]: can't you inject the manager here?

File: sign-core/src/main/java/eu/cloudnetservice/sign/core/packets/PacketInSignSelector.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public final class PacketInSignSelector extends PacketInHandler {
+	@Override
+	public void handleInput(Document data, PacketSender packetSender) {
```
Comment (GiantTreeLP) [inline sign-core/src/main/java/eu/cloudnetservice/sign/core/packets/PacketInSignSelector.java]: this whole method looks more like it belongs to the manager itself

File: sign-module/src/main/java/eu/cloudnetservice/sign/module/ProjectMain.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+import de.dytanic.cloudnetcore.api.event.network.UpdateAllEvent;
+import de.dytanic.cloudnetcore.network.components.MinecraftServer;
+import de.dytanic.cloudnetcore.network.components.Wrapper;
+import eu.cloudnetservice.sign.module.config.ConfigSignLayout;
+import eu.cloudnetservice.sign.module.database.SignDatabase;
+import eu.cloudnetservice.sign.module.packet.in.PacketInAddSign;
+import eu.cloudnetservice.sign.module.packet.in.PacketInRemoveSign;
+import eu.cloudnetservice.sign.module.packet.out.PacketOutSignSelector;
+
+public class ProjectMain extends CoreModule implements IEventListener<UpdateAllEvent> {
```
Comment (GiantTreeLP) [inline sign-module/src/main/java/eu/cloudnetservice/sign/module/ProjectMain.java]: I don't like this name

File: sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/ProjectMain.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+import de.dytanic.cloudnet.api.CloudAPI;
+import de.dytanic.cloudnet.lib.network.protocol.packet.PacketRC;
+import eu.cloudnetservice.sign.core.manager.SignManager;
+import eu.cloudnetservice.sign.core.packets.PacketInSignSelector;
+import eu.cloudnetservice.sign.plugin.commands.SignsCommand;
+import eu.cloudnetservice.sign.plugin.manager.BukkitSignManager;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public final class ProjectMain extends JavaPlugin {
```
Comment (GiantTreeLP) [inline sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/ProjectMain.java]: As you can guess, this name is not very much saying anything

File: sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/ProjectMain.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+	private BukkitSignManager signManager;
+
+	@Override
+	public void onEnable() {
+		System.out.println("Initiate bukkit sign manager");
```
Comment (GiantTreeLP) [inline sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/ProjectMain.java]: Please use the logger for ANY output

File: sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+				try {
+					for (Sign sign : SignManager.getInstance().getSigns().values()) {
+						if (Bukkit.getWorld(sign.getPosition().getWorld()) != null) {
+							Location location = signNetworkHandlerAdapter.toLocation(sign.getPosition());
+							for (Entity entity : location.getWorld().getNearbyEntities(location,
+									SignManager.getInstance().getSignLayoutConfig().getDistance(),
+									SignManager.getInstance().getSignLayoutConfig().getDistance(),
+									SignManager.getInstance().getSignLayoutConfig().getDistance())) {
+								if (entity instanceof Player && !entity.hasPermission("cloudnet.signs.knockback.bypass")) {
+									Bukkit.getScheduler().runTask(signNetworkHandlerAdapter.getPlugin(), new Runnable() {
```
Comment (GiantTreeLP) [inline sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java]: please use lambdas or method references

File: sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+					for (Sign sign : SignManager.getInstance().getSigns().values()) {
+						boolean exists = signNetworkHandlerAdapter.exists(sign);
+
+						if (!exists) {
+							sign.setServerInfo(null);
+							continue;
+						}
+
+						if (signNetworkHandlerAdapter.isMaintenance(sign.getTargetGroup())) {
+							SignLayout _signLayout = signNetworkHandlerAdapter.getLayout(sign.getTargetGroup(), "maintenance");
```
Comment (GiantTreeLP) [inline sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java]: this name doesn't say anything, why it is special. what does the underscore even mean?

File: sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.util.Vector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ThreadImpl extends Thread {
```
Comment (GiantTreeLP) [inline sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java]: Why is this a thread and not a normal task?

File: sign-plugin/src/main/resources/plugin.yml [hunk 14]
```diff
@@ -0,0 +1,13 @@
+name: SignSystem
+version: ${sign.version}
+author: TheMeinerLP
+website: https://cloudnetservice.eu/
+description: CloudNet Sign Interface for the Central server
+
+main: eu.cloudnetservice.sign.plugin.ProjectMain
+
+softdepend: [CloudNetAPI]
```
Comment (GiantTreeLP) [inline sign-plugin/src/main/resources/plugin.yml]: With how this plugin is made, this is NOT a soft dependency, it is very much hard.

**Review by GiantTreeLP — CHANGES_REQUESTED** [review]
There is room for improvement. So much so, that it's probably worth it to redo the entire plugin.
There are many things that need renaming or adjustments to their interfaces and methods.

File: sign-core/pom.xml (hunk 1 above)
Comment (TheMeinerLP) [reply sign-core/pom.xml:10]: No, is a module get the groupId from parent

**Review by GiantTreeLP — APPROVED** [review]

**Pull request merged**